### PR TITLE
Issue #2047 Hyper-V Admin group check

### PR DESCRIPTION
**Pull request by paul-green**
Potential fix for Hyper-V Admin group check (e.g. #1541)

### CODE REVIEW

**Comment by anjannath** [comment]
You could have fixed the commit message in #2050

**Comment by paul-green** [comment]
I checked not long ago and the pr said its behind master, so i clicked on Update so i guess this has done a merge.  Shall i revert?

Also i tried to fix the commit message by making a whitespace commit, but the verify commit task still failed as the previous commit message was still there.

**Comment by anjannath** [comment]
you can amend your old commit using `git commit --amend` Please close this PR and update #2050 That has more information.

**Comment by paul-green** [comment]
Ok

**Comment by anjannath** [comment]
@paul-green please update it soon so we can merge #2050 ASAP.

**Comment by paul-green** [comment]
I'll fix as soon as i get home

**Comment by paul-green** [comment]
I'm not able to re-open 2050 PR - Reopen button is disabled. Pls advise. Thanks

**Comment by anjannath** [comment]
PTAL at https://github.com/minishift/minishift/pull/2050#issuecomment-367287180

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -374,7 +374,12 @@ func checkHypervDriverUser() bool {
 	// Use RID to prevent issues with localized groups: https://github.com/minishift/minishift/issues/1541
 	// https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems
 	// BUILTIN\Hyper-V Administrators => S-1-5-32-578
-	checkIfMemberOfHyperVAdmins := `@([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole("S-1-5-32-578")`
+
+	//Hyper-V Administrators group check fails: https://github.com/minishift/minishift/issues/2047
+	//Using SecurityIdentifier overload of IsInRole()
+	checkIfMemberOfHyperVAdmins :=
+		`$sid = New-Object System.Security.Principal.SecurityIdentifier("S-1-5-32-578")
+	@([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole($sid)`
```
Comment (LalatenduMohanty) [inline cmd/minishift/cmd/start_preflight.go]: @paul-green Did you check if this works for non English users too?

Comment (paul-green) [reply cmd/minishift/cmd/start_preflight.go]: No - I don't have the resources to do that

**Comment by anjannath** [comment]
Working for english and non english localized versions of windows. :+1:

**Comment by paul-green** [comment]
Will do

**Comment by paul-green** [comment]
@anjannath Commit message updated to include ref to #1882

File: cmd/minishift/cmd/start_preflight.go [hunk 2]
```diff
@@ -374,7 +374,12 @@ func checkHypervDriverUser() bool {
 	// Use RID to prevent issues with localized groups: https://github.com/minishift/minishift/issues/1541
 	// https://support.microsoft.com/en-us/help/243330/well-known-security-identifiers-in-windows-operating-systems
 	// BUILTIN\Hyper-V Administrators => S-1-5-32-578
-	checkIfMemberOfHyperVAdmins := `@([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole("S-1-5-32-578")`
+
+	//Hyper-V Administrators group check fails: https://github.com/minishift/minishift/issues/2047
+	//Using SecurityIdentifier overload of IsInRole()
+	checkIfMemberOfHyperVAdmins :=
+		`$sid = New-Object System.Security.Principal.SecurityIdentifier("S-1-5-32-578")
```
Comment (anjannath) [inline cmd/minishift/cmd/start_preflight.go]: :+1:

**Comment by LalatenduMohanty** [comment]
Thanks @paul-green.

**Comment by LalatenduMohanty** [comment]
@anjannath @paul-green The commit message should have been `Issue #2047 #1882 Fix Hyper-V Admin group check`. I missed to cross check when merging the PR. It is not very bad but could have been better.

**Pull request merged**